### PR TITLE
[Mobile] Add Cover block overlay color settings

### LINKED ISSUE #1735: Cover block: Overlay color settings
Depends on 
[Add Cover Block - 1st iteration](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1734)
[Add Color picker to be used in block settings](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1692)

We need to be able to set Overlay color of the cover block.

### LINKED ISSUE #1737: Cover block: Overlay gradient settings
Depends on 
[Add Cover Block - 1st iteration](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1734)
[Add Gradient picker to be used in block settings](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1736)

We need to be able to set "Overlay gradient" of the Cover block.

### PR DESCRIPTION
**Pull request by mkevins**
### Related PR

`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2242

## Description

This PR adds overlay color and gradient settings to the mobile Cover block. This resolves https://github.com/wordpress-mobile/gutenberg-mobile/issues/1735, and partially resolves https://github.com/wordpress-mobile/gutenberg-mobile/issues/1737.

## To test
1. In a new post, add a Cover block
2. Select an image from the media library
3. Tap the up-arrow on the floating toolbar (to make sure the Cover block is selected, not the inner paragraph block)
4. Tap the settings gear
5. Tap "Overlay color" 
6. Change the color in the following ways:
   1. Pick a pre-selected color from the palette
   2. Choose a custom color (scroll all the way to the right)
   3. Choose a pre-selected gradient (tap the gradient option on the bottom)

### Expected

Colors / gradients should update the block as they are selected.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/cover/overlay-color-settings.native.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+	const colorValue = getColorObjectByAttributeValues(
+		colors,
+		overlayColor,
+		customOverlayColor
+	).color;
+
+	const setOverlayAttribute = ( attributeName, value ) => {
+		setAttributes( {
+			...attributes,
```
Comment (geriux) [inline packages/block-library/src/cover/overlay-color-settings.native.js:41]: Just wondering why do we need to spread the current attributes 🤔I don't think we need that, what do you think?

**Comment by geriux** [comment]
Tested it and its looking great! 🤩 just left one comment https://github.com/WordPress/gutenberg/pull/22207#discussion_r423010273

File: packages/block-library/src/cover/overlay-color-settings.native.js (hunk 1 above)
Comment (mkevins) [reply packages/block-library/src/cover/overlay-color-settings.native.js]: You're right, that's not necessary :smile:, nice catch! I've removed it.

**Review by geriux — APPROVED** [review]
LGTM! Nice work! 🎨

**Comment by mkevins** [comment]
Thanks for testing Gerardo!

**Pull request merged**